open-metadata-resources/open-metadata-demos/virtual-data-connector/src/main/java/v1/eventmapper/IGCEventProcessor.java

```diff
@@ -0,0 +1,99 @@
+import org.odpi.openmetadata.repositoryservices.eventmanagement.OMRSRepositoryEventManager;
+import org.odpi.openmetadata.repositoryservices.ffdc.exception.TypeErrorException;
+import org.odpi.openmetadata.repositoryservices.connectors.stores.metadatacollectionstore.properties.instances.*;
+
+
+public class IGCEventProcessor {
+
+    private  IGCColumn igcColumn;
+    private  String originatorServerName;
+    private IGCOMRSRepositoryConnector igcomrsRepositoryConnector;
+    private  OMRSRepositoryEventManager omrsRepositoryEventManager;
+    private  String businessTerm;
+
+
+    public  void process(IGCKafkaEvent igcKafkaEvent) {
+        if (igcKafkaEvent.getASSETTYPE().equals("Database Column") &&
+                (
+                igcKafkaEvent.getACTION().equals("ASSIGNED_RELATIONSHIP") || igcKafkaEvent.getACTION().equals("MODIFY")
+                )
+            ) {
+            try {
+                igcColumn = igcomrsRepositoryConnector.queryIGCColumn(igcKafkaEvent.getASSETRID());
+                originatorServerName = igcColumn.getName();
+                if (igcColumn.getAssignedToTerms() != null) {
+                    businessTerm = igcColumn.getAssignedToTerms().getItems().get(0).getName();
+                    Relationship relationship = newRelationship("SemanticAssignment", igcKafkaEvent.getASSETRID(), "RelationalColumn", igcColumn.getAssignedToTerms().getItems().get(0).getId(), "GlossaryTerm");
+                    omrsRepositoryEventManager.processNewRelationshipEvent(
+                            "IGCOMRSRepositoryEventMapper",
+                            igcomrsRepositoryConnector.getMetadataCollectionId(),
+                            originatorServerName,
+                            "IGC",
+                            "",
+                            relationship
+                    );
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private  Relationship newRelationship(String relationshipType, String entityId1,String entityType1, String entityId2, String entityType2) throws TypeErrorException {
+        //Set Relationship linking technical term with IGC business term
+        Relationship relationship = initiateRelationship(relationshipType);
+
+        EntityProxy entityProxyOne = newEntityProxy(entityType1);//TODO
+        EntityProxy entityProxyTwo = newEntityProxy(entityType2);
+
+        entityProxyOne.setGUID(entityId1);
+        entityProxyTwo.setGUID(entityId2);
+
+        relationship.setEntityOnePropertyName(originatorServerName);
+        relationship.setEntityOneProxy(entityProxyOne);
+
+
+        relationship.setEntityTwoPropertyName(businessTerm);
+        relationship.setEntityTwoProxy(entityProxyTwo);
+        return relationship;
+    }
+
+    private  Relationship initiateRelationship(String relationshipType) throws TypeErrorException {
+        return new OMRSRepositoryContentHelper().getNewRelationship(
```

Comment:
Should be using the repository content helper passed to the repository event mapper on the initialize request.